### PR TITLE
Update ConfigImport

### PR DESCRIPTION
**Pull request by nitinsinghoxidesales**
We have noticed some issues with OXID Shop 6.2.X
**Issues**:
1. Config import is not enabling a module if it's deactivated in DB. 
2. Config import is not creating sub-shops through configuration files. 

**Changes**:
1. Checking whether module is already active before deactivating the module. 

> $oModuleStateFixer->deactivate($oModule)

2. Creating sub-shops through defined configuration files.

**TODO** 
1. Need to test of Shop 6.1 (Currently checks are failing for it)
2. Add some more UT test cases

### CODE REVIEW

**Comment by nitinsinghoxidesales** [comment]
Hi @keywan-ghadami-oxid @alfredbez - Can you please check this PR and share your inputs if you think we can improve it more?

Note: Still there are 2 TODO points need to implement.

File: Controller/Admin/ShopMain.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+     * Check if Shop can be created.
+     *
+     * @param string                                   $shopId
+     * @param \OxidEsales\Eshop\Application\Model\Shop $shop
+     *
+     * @return bool
+     */
+    public function canCreateShop($shopId, $shop)
+    {
+        return parent::canCreateShop($shopId, $shop);
```
Comment (alfredbez) [inline Controller/Admin/ShopMain.php:36]: I don't get the point of this class

Comment (nitinsinghoxidesales) [reply Controller/Admin/ShopMain.php:34]: Ahh, Sorry i missed your comment. 

Actually i want to call **canCreateShop** method in our command. but this method is protected. Hence, i am just overriding this method and making it public. If you have any better suggestion then please let me know. 
phttps://github.com/OXID-eSales/oxideshop_ce/blob/master/source/Application/Controller/Admin/ShopMain.php#L290
https://github.com/OXID-eSales/oxideshop_ee/blob/master/Application/Controller/Admin/ShopMain.php#L121

**Comment by nitinsinghoxidesales** [comment]
@alfredbez @keywan-ghadami-oxid - Can you please review and merge this long pending PR?

Please let me know if you have any suggestion on this.

**Comment by alfredbez** [comment]
I can't do this in the next weeks

**Comment by nitinsinghoxidesales** [comment]
> I can't do this in the next weeks

:( 
May be @keywan-ghadami-oxid would like to review and merge?

**Comment by alfredbez** [comment]
Looks like there are still some open todos from your initial comment on this PR, did you already checked them?
> 1. Need to test of Shop 6.1 (Currently checks are failing for it)
> 2. Add some more UT test cases

**Comment by nitinsinghoxidesales** [comment]
Yes, I have checked today. 

1. Shop 6.1 is now green. https://github.com/OXIDprojects/oxid_modules_config/pull/44/checks. Earlier it was failing because of DB connection issue. 
2. Adding more UT test cases - I wanted to add some more but we have some time constraint. It is working in the checks with earlier UT.  Do you suggest to add more UT test cases?

**Comment by nitinsinghoxidesales** [comment]
@keywan-ghadami-oxid  - Can you please review it and merge it?

**Pull request merged**